### PR TITLE
cleanup: refactor bloated rehype plugin entry file and fix tech debts

### PR DESCRIPTION
**Pull request by imballinst**
Prerequisites: https://github.com/imballinst/protosaurus/pull/34 and https://github.com/imballinst/protosaurus/pull/35.

This PR contains mostly cleanups of the bloated `packages/rehype-plugin-codeblock/src/index.ts` file, as well as doing some of the `TODO`s that previously weren't touched.

### CODE REVIEW

File: packages/rehype-plugin-codeblock/src/hast/parser.test.ts [hunk 1]
```diff
@@ -179,41 +179,3 @@ describe('stripTitleFromElementProperties', () => {
...
-    // Swap the positions.
-    expect(
-      stripTitleFromElementProperties({
-        collapsible: true,
-        title: '"Hello',
-        'World"': true
-      })
-    ).to.eql({ collapsible: true });
-  });
-});
```
Comment (imballinst) [inline packages/rehype-plugin-codeblock/src/hast/parser.test.ts:219]: This is a duplicate test, not sure why it's here...

File: packages/rehype-plugin-codeblock/src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Comment, DocType, Element, Text } from 'hast-format';
```
Comment (dio) [inline packages/rehype-plugin-codeblock/src/helpers.ts:17]: Sorry seems like we need to activate license checking on CI. But, meanwhile, can we have it here? Thank you!

File: packages/rehype-plugin-codeblock/src/hast/annotations.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import { Element, Text } from 'hast-format';
```
Comment (dio) [inline packages/rehype-plugin-codeblock/src/hast/annotations.ts:17]: Also license.

**Review by dio — APPROVED** [review]

**Comment by imballinst** [comment]
Ooo, yes, I totally forgot about the license headers. I'm on it. Thanks for catching it!

**Comment by imballinst** [comment]
Thanks @dio for the review! I'll work on the CI check for license in another PR.

**Pull request merged**